In [12]:
import pandas as pd
import numpy as np
from scipy import stats

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [13]:
df1 = pd.read_csv('./마지막/롯데월드타워(잠실역2번출구 쪽) 365일.csv', encoding = 'cp949')
df2 = pd.read_csv('./마지막/올림픽공원역 3번출구 365일.csv', encoding = 'cp949')
df3 = pd.read_csv('./마지막/잠실역 8번출구 365일.csv', encoding = 'cp949')
df4 = pd.read_csv('./마지막/한성백제역 2번 출구 365일.csv', encoding = 'cp949')
df5 = pd.read_csv('./마지막/몽촌토성역 1번출구 옆 365일.csv', encoding = 'cp949')
df6 = pd.read_csv('./마지막/방이삼거리 365일.csv', encoding = 'cp949')
df7 = pd.read_csv('./마지막/방이역 4번출구 365일.csv', encoding = 'cp949')
df8 = pd.read_csv('./마지막/삼전사거리 포스코더샵 365일.csv', encoding = 'cp949')
df9 = pd.read_csv('./마지막/석촌고분역 2번 출구 365일.csv', encoding = 'cp949')
df10 = pd.read_csv('./마지막/석촌동 주민센터 365일.csv', encoding = 'cp949')
df11 = pd.read_csv('./마지막/석촌역 2번출구 365일.csv', encoding = 'cp949')
df12 = pd.read_csv('./마지막/석촌역 8번출구 365일.csv', encoding = 'cp949')
df13 = pd.read_csv('./마지막/석촌호교차로 (스타벅스 앞) 365일.csv', encoding = 'cp949')
df14 = pd.read_csv('./마지막/문정 법조단지8 365일.csv', encoding = 'cp949')
df15 = pd.read_csv('./마지막/문정동 힐스데이트 에코 송파 365일.csv', encoding = 'cp949')
df16 = pd.read_csv('./마지막/문정 법조단지7 365일.csv', encoding = 'cp949')
df17 = pd.read_csv('./마지막/석촌호수 서호사거리 365일.csv', encoding = 'cp949')
df18 = pd.read_csv('./마지막/석촌호수교차로(동호 팔각정 앞) 365일.csv', encoding = 'cp949')
df19 = pd.read_csv('./마지막/송파구청 교차로 365일.csv', encoding = 'cp949')
df20 = pd.read_csv('./마지막/송파나루역 4번 출구옆 365일.csv', encoding = 'cp949')
df21 = pd.read_csv('./마지막/송파역 4번출구 365일.csv', encoding = 'cp949')
df22 = pd.read_csv('./마지막/송파지역자활센터 뒤 365일.csv', encoding = 'cp949')
df23 = pd.read_csv('./마지막/여흥레이크빌 앞 (석촌호수 까페거리) 365일.csv', encoding = 'cp949')
df24 = pd.read_csv('./마지막/잠실나들목 365일.csv', encoding = 'cp949')
df25 = pd.read_csv('./마지막/잠실나들목4 365일.csv', encoding = 'cp949')
df26 = pd.read_csv('./마지막/잠실나루역 (2번 출구 쪽) 365일.csv', encoding = 'cp949')
df27 = pd.read_csv('./마지막/잠실본동주민센터뒤 잠실근린공원 365일.csv', encoding = 'cp949')
df28 = pd.read_csv('./마지막/잠실새내역 5번 출구 뒤 365일.csv', encoding = 'cp949')
df29 = pd.read_csv('./마지막/잠실역 6번출구 365일.csv', encoding = 'cp949')
df30 = pd.read_csv('./마지막/천호역 10번 출구 앞 365일.csv', encoding = 'cp949')

In [14]:
df = pd.merge(df1, df2, how = 'outer')
df = pd.merge(df, df3, how = 'outer')
df = pd.merge(df, df4, how = 'outer')
df = pd.merge(df, df5, how = 'outer')
df = pd.merge(df, df6, how = 'outer')
df = pd.merge(df, df7, how = 'outer')
df = pd.merge(df, df8, how = 'outer')
df = pd.merge(df, df9, how = 'outer')
df = pd.merge(df, df10, how = 'outer')
df = pd.merge(df, df11, how = 'outer')
df = pd.merge(df, df12, how = 'outer')
df = pd.merge(df, df13, how = 'outer')
df = pd.merge(df, df14, how = 'outer')
df = pd.merge(df, df15, how = 'outer')
df = pd.merge(df, df16, how = 'outer')
df = pd.merge(df, df17, how = 'outer')
df = pd.merge(df, df18, how = 'outer')
df = pd.merge(df, df19, how = 'outer')
df = pd.merge(df, df20, how = 'outer')
df = pd.merge(df, df21, how = 'outer')
df = pd.merge(df, df22, how = 'outer')
df = pd.merge(df, df23, how = 'outer')
df = pd.merge(df, df24, how = 'outer')
df = pd.merge(df, df25, how = 'outer')
df = pd.merge(df, df26, how = 'outer')
df = pd.merge(df, df27, how = 'outer')
df = pd.merge(df, df28, how = 'outer')
df = pd.merge(df, df29, how = 'outer')
df = pd.merge(df, df30, how = 'outer')
df

자치구               대여소명     년  월   일  시간  lambda    mu      p_1+  \
0       송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1   0     0.0   1.0  0.000000   
1       송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1   1     0.0   0.0  0.000000   
2       송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1   2     0.0   0.0  0.000000   
3       송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1   3     0.0   0.0  0.000000   
4       송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1   4     1.0   0.0  0.000000   
...     ...                ...   ... ..  ..  ..     ...   ...       ...   
262795  송파구       천호역 10번 출구 앞  2021  5  31  19     8.0  17.0  0.470306   
262796  송파구       천호역 10번 출구 앞  2021  5  31  20     9.0   8.0  0.944378   
262797  송파구       천호역 10번 출구 앞  2021  5  31  21     7.0   8.0  0.830376   
262798  송파구       천호역 10번 출구 앞  2021  5  31  22     6.0   7.0  0.818234   
262799  송파구       천호역 10번 출구 앞  2021  5  31  23     2.0   1.0  0.999022   

             L_s  재고량  3사분위수  2사분위수  1사분위수  균형 내 균형  
0       0.000000  0.0   16.0    6.0    0.0  균형  불균형  
1       0.000000  0.0   16.0    6.0    0.0  균형  불균형  
2       0.000000  0.0   16.0    6.0    0.0  균형  불균형  
3       0.000000  0.0   16.0    6.0    0.0  균형  불균형  
4       0.000000  1.0   16.0    6.0    0.0  균형  불균형  
...          ...  ...    ...    ...    ...  ..  ...  
262795  0.883560  4.0   11.0    6.0    3.0  균형  불균형  
262796  5.449743  5.0   11.0    6.0    3.0  균형   균형  
262797  3.430087  4.0   11.0    6.0    3.0  균형   균형  
262798  3.276410  3.0   11.0    6.0    3.0  균형   균형  
262799  8.009775  4.0   11.0    6.0    3.0  균형   균형  

[262800 rows x 16 columns]

In [15]:
df_new = df.drop(['자치구', 'lambda', 'mu', '2사분위수'], axis = 1)
df_new = df_new.rename(columns = {'균형' : '기존균형', '내 균형' : 'Queuing균형'})
df_new

대여소명     년  월   일  시간      p_1+       L_s  재고량  3사분위수  \
0       롯데월드타워(잠실역2번출구 쪽)  2020  6   1   0  0.000000  0.000000  0.0   16.0   
1       롯데월드타워(잠실역2번출구 쪽)  2020  6   1   1  0.000000  0.000000  0.0   16.0   
2       롯데월드타워(잠실역2번출구 쪽)  2020  6   1   2  0.000000  0.000000  0.0   16.0   
3       롯데월드타워(잠실역2번출구 쪽)  2020  6   1   3  0.000000  0.000000  0.0   16.0   
4       롯데월드타워(잠실역2번출구 쪽)  2020  6   1   4  0.000000  0.000000  1.0   16.0   
...                   ...   ... ..  ..  ..       ...       ...  ...    ...   
262795       천호역 10번 출구 앞  2021  5  31  19  0.470306  0.883560  4.0   11.0   
262796       천호역 10번 출구 앞  2021  5  31  20  0.944378  5.449743  5.0   11.0   
262797       천호역 10번 출구 앞  2021  5  31  21  0.830376  3.430087  4.0   11.0   
262798       천호역 10번 출구 앞  2021  5  31  22  0.818234  3.276410  3.0   11.0   
262799       천호역 10번 출구 앞  2021  5  31  23  0.999022  8.009775  4.0   11.0   

        1사분위수 기존균형 Queuing균형  
0         0.0   균형       불균형  
1         0.0   균형       불균형  
2         0.0   균형       불균형  
3         0.0   균형       불균형  
4         0.0   균형       불균형  
...       ...  ...       ...  
262795    3.0   균형       불균형  
262796    3.0   균형        균형  
262797    3.0   균형        균형  
262798    3.0   균형        균형  
262799    3.0   균형        균형  

[262800 rows x 12 columns]

In [16]:
df_new[(df_new['월'] == 5) & (df_new['일'] == 1) & (df_new['시간'] == 18)].reset_index(drop = True)

대여소명     년  월  일  시간      p_1+        L_s   재고량  3사분위수  \
0      롯데월드타워(잠실역2번출구 쪽)  2021  5  1  18  0.444444   0.800000  22.0   16.0   
1            올림픽공원역 3번출구  2021  5  1  18  0.000000   0.000000  21.0    8.0   
2               잠실역 8번출구  2021  5  1  18  0.998940  16.063626   8.0   10.0   
3            한성백제역 2번 출구  2021  5  1  18  0.950000   9.500000   6.0   13.0   
4           몽촌토성역 1번출구 옆  2021  5  1  18  0.999661  15.012188   8.0   10.0   
5                  방이삼거리  2021  5  1  18  0.821765   3.958841   7.0   14.0   
6               방이역 4번출구  2021  5  1  18  0.000000   0.000000   4.0    9.0   
7            삼전사거리 포스코더샵  2021  5  1  18  0.916667   5.500000  13.0   23.0   
8            석촌고분역 2번 출구  2021  5  1  18  0.657764   1.759637  20.0   18.0   
9               석촌동 주민센터  2021  5  1  18  0.888889   4.000000  36.0   15.0   
10              석촌역 2번출구  2021  5  1  18  0.250000   0.333333   6.0    7.0   
11              석촌역 8번출구  2021  5  1  18  0.998043   7.017613   9.0    6.0   
12       석촌호교차로 (스타벅스 앞)  2021  5  1  18  0.989096   8.392551  16.0   14.0   
13              문정 법조단지8  2021  5  1  18  0.916667   5.500000   4.0    5.0   
14       문정동 힐스데이트 에코 송파  2021  5  1  18  0.000000   0.000000  11.0    8.0   
15              문정 법조단지7  2021  5  1  18  0.664078   1.906793   9.0    6.0   
16            석촌호수 서호사거리  2021  5  1  18  0.999999  10.600004  13.0   16.0   
17     석촌호수교차로(동호 팔각정 앞)  2021  5  1  18  0.000000   0.000000   5.0    7.0   
18              송파구청 교차로  2021  5  1  18  0.991176   7.176476  12.0    9.0   
19          송파나루역 4번 출구옆  2021  5  1  18  0.666441   1.987812  48.0   19.0   
20              송파역 4번출구  2021  5  1  18  0.999756  10.002930   7.0   12.0   
21            송파지역자활센터 뒤  2021  5  1  18  0.916667   5.500000  51.0   29.0   
22  여흥레이크빌 앞 (석촌호수 까페거리)  2021  5  1  18  0.333332   0.499977   4.0   10.0   
23                 잠실나들목  2021  5  1  18  0.499996   0.999870  45.0   22.0   
24                잠실나들목4  2021  5  1  18  0.999996  16.000069  18.0   11.0   
25       잠실나루역 (2번 출구 쪽)  2021  5  1  18  0.666441   1.987812   7.0   10.0   
26      잠실본동주민센터뒤 잠실근린공원  2021  5  1  18  0.000000   0.000000  29.0   24.0   
27         잠실새내역 5번 출구 뒤  2021  5  1  18  0.665904   1.965667   7.0   12.0   
28              잠실역 6번출구  2021  5  1  18  0.000000   0.000000   2.0    9.0   
29          천호역 10번 출구 앞  2021  5  1  18  0.999997   8.666676   5.0   11.0   

    1사분위수 기존균형 Queuing균형  
0    0.00  불균형       불균형  
1    1.00  불균형       불균형  
2   -3.00   균형       불균형  
3    3.00   균형        균형  
4    1.75   균형       불균형  
5    2.00   균형        균형  
6    2.00   균형       불균형  
7    6.00   균형        균형  
8    4.00  불균형        균형  
9    3.00  불균형        균형  
10   1.00   균형       불균형  
11   1.00  불균형        균형  
12   3.00  불균형        균형  
13   1.00   균형       불균형  
14   0.00  불균형       불균형  
15  -3.00  불균형        균형  
16   3.00   균형        균형  
17   1.00   균형       불균형  
18   0.00  불균형        균형  
19   4.00  불균형        균형  
20   3.00   균형        균형  
21   8.00  불균형        균형  
22   2.00   균형       불균형  
23   7.00  불균형       불균형  
24   0.00  불균형       불균형  
25   2.00   균형        균형  
26   7.00  불균형       불균형  
27   3.00   균형        균형  
28   2.00   균형       불균형  
29   3.00   균형        균형

# 월별

In [5]:
df_new_e = df_new.copy()

In [6]:
for i in range(1, 13):
    df_new_e = df_new[df_new['월'] == i]
    ad_cross = pd.crosstab(df_new_e['기존균형'], df_new_e['Queuing균형'])
    chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
    print(f'chi2  = {chi2:.2f}(df={dof:.2f}), p value > {p:.5f}.')

chi2  = 9.04(df=1.00), p value > 0.00265.
chi2  = 43.04(df=1.00), p value > 0.00000.
chi2  = 0.98(df=1.00), p value > 0.32278.
chi2  = 14.57(df=1.00), p value > 0.00014.
chi2  = 10.83(df=1.00), p value > 0.00100.
chi2  = 16.39(df=1.00), p value > 0.00005.
chi2  = 170.50(df=1.00), p value > 0.00000.
chi2  = 143.98(df=1.00), p value > 0.00000.
chi2  = 9.76(df=1.00), p value > 0.00179.
chi2  = 3.21(df=1.00), p value > 0.07323.
chi2  = 3.14(df=1.00), p value > 0.07648.
chi2  = 104.42(df=1.00), p value > 0.00000.


# 전체

In [17]:
ad_cross = pd.crosstab(df_new['기존균형'], df_new['Queuing균형'])
ad_cross

Queuing균형     균형    불균형
기존균형                   
균형         54559  92455
불균형        45205  70581

In [19]:
chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
chi2, p, dof

(102.4913082121348, 4.332644621466487e-24, 1)

In [20]:
print(f'chi2  = {chi2:.2f}(df={dof:.2f}), p value > {p:.5f}.')

chi2  = 102.49(df=1.00), p value > 0.00000.


# 시간별

In [10]:
for i in range(24):
    df_hour = df_new[(df_new['시간'] == i)]
    ad_cross = pd.crosstab(df_hour['기존균형'],  df_hour['Queuing균형'])
    chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
    print(f'chi2  = {chi2:.2f}(df={dof:.5f}), p value > {p:.5f}.')

chi2  = 6.28(df=1.00000), p value > 0.01221.
chi2  = 0.58(df=1.00000), p value > 0.44633.
chi2  = 0.07(df=1.00000), p value > 0.79675.
chi2  = 1.42(df=1.00000), p value > 0.23350.
chi2  = 0.06(df=1.00000), p value > 0.80437.
chi2  = 0.16(df=1.00000), p value > 0.69179.
chi2  = 7.46(df=1.00000), p value > 0.00631.
chi2  = 13.34(df=1.00000), p value > 0.00026.
chi2  = 0.52(df=1.00000), p value > 0.46943.
chi2  = 2.84(df=1.00000), p value > 0.09197.
chi2  = 25.09(df=1.00000), p value > 0.00000.
chi2  = 4.86(df=1.00000), p value > 0.02750.
chi2  = 5.02(df=1.00000), p value > 0.02500.
chi2  = 6.27(df=1.00000), p value > 0.01225.
chi2  = 10.35(df=1.00000), p value > 0.00129.
chi2  = 2.23(df=1.00000), p value > 0.13521.
chi2  = 1.70(df=1.00000), p value > 0.19249.
chi2  = 0.00(df=1.00000), p value > 0.98874.
chi2  = 0.01(df=1.00000), p value > 0.92401.
chi2  = 39.58(df=1.00000), p value > 0.00000.
chi2  = 65.06(df=1.00000), p value > 0.00000.
chi2  = 63.93(df=1.00000), p value > 0.00000.
chi2

In [8]:
df1 = pd.read_csv('./최종/롯데월드타워(잠실역2번출구 쪽).csv', encoding = 'cp949')
df2 = pd.read_csv('./최종/올림픽공원역 3번출구.csv', encoding = 'cp949')
df3 = pd.read_csv('./최종/잠실역 8번출구.csv', encoding = 'cp949')
df4 = pd.read_csv('./최종/한성백제역 2번 출구.csv', encoding = 'cp949')
df5 = pd.read_csv('./최종/몽촌토성역 1번출구 옆.csv', encoding = 'cp949')
df6 = pd.read_csv('./최종/방이삼거리.csv', encoding = 'cp949')
df7 = pd.read_csv('./최종/방이역 4번출구.csv', encoding = 'cp949')
df8 = pd.read_csv('./최종/삼전사거리 포스코더샵.csv', encoding = 'cp949')
df9 = pd.read_csv('./최종/석촌고분역 2번 출구.csv', encoding = 'cp949')
df10 = pd.read_csv('./최종/석촌동 주민센터.csv', encoding = 'cp949')
df11 = pd.read_csv('./최종/석촌역 2번출구.csv', encoding = 'cp949')
df12 = pd.read_csv('./최종/석촌역 8번출구.csv', encoding = 'cp949')
df13 = pd.read_csv('./최종/석촌호교차로 (스타벅스 앞).csv', encoding = 'cp949')
df14 = pd.read_csv('./최종/문정 법조단지8.csv', encoding = 'cp949')
df15 = pd.read_csv('./최종/문정동 힐스데이트 에코 송파.csv', encoding = 'cp949')
df16 = pd.read_csv('./최종/문정 법조단지7.csv', encoding = 'cp949')
df17 = pd.read_csv('./최종/석촌호수 서호사거리.csv', encoding = 'cp949')
df18 = pd.read_csv('./최종/석촌호수교차로(동호 팔각정 앞).csv', encoding = 'cp949')
df19 = pd.read_csv('./최종/송파구청 교차로.csv', encoding = 'cp949')
df20 = pd.read_csv('./최종/송파나루역 4번 출구옆.csv', encoding = 'cp949')
df21 = pd.read_csv('./최종/송파역 4번출구.csv', encoding = 'cp949')
df22 = pd.read_csv('./최종/송파지역자활센터 뒤.csv', encoding = 'cp949')
df23 = pd.read_csv('./최종/여흥레이크빌 앞 (석촌호수 까페거리).csv', encoding = 'cp949')
df24 = pd.read_csv('./최종/잠실나들목.csv', encoding = 'cp949')
df25 = pd.read_csv('./최종/잠실나들목4.csv', encoding = 'cp949')
df26 = pd.read_csv('./최종/잠실나루역 (2번 출구 쪽).csv', encoding = 'cp949')
df27 = pd.read_csv('./최종/잠실본동주민센터뒤 잠실근린공원.csv', encoding = 'cp949')
df28 = pd.read_csv('./최종/잠실새내역 5번 출구 뒤.csv', encoding = 'cp949')
df29 = pd.read_csv('./최종/잠실역 6번출구.csv', encoding = 'cp949')
df30 = pd.read_csv('./최종/천호역 10번 출구 앞.csv', encoding = 'cp949')

In [9]:
df = pd.merge(df1, df2, how = 'outer')
df = pd.merge(df, df3, how = 'outer')
df = pd.merge(df, df4, how = 'outer')
df = pd.merge(df, df5, how = 'outer')
df = pd.merge(df, df6, how = 'outer')
df = pd.merge(df, df7, how = 'outer')
df = pd.merge(df, df8, how = 'outer')
df = pd.merge(df, df9, how = 'outer')
df = pd.merge(df, df10, how = 'outer')
df = pd.merge(df, df11, how = 'outer')
df = pd.merge(df, df12, how = 'outer')
df = pd.merge(df, df13, how = 'outer')
df = pd.merge(df, df14, how = 'outer')
df = pd.merge(df, df15, how = 'outer')
df = pd.merge(df, df16, how = 'outer')
df = pd.merge(df, df17, how = 'outer')
df = pd.merge(df, df18, how = 'outer')
df = pd.merge(df, df19, how = 'outer')
df = pd.merge(df, df20, how = 'outer')
df = pd.merge(df, df21, how = 'outer')
df = pd.merge(df, df22, how = 'outer')
df = pd.merge(df, df23, how = 'outer')
df = pd.merge(df, df24, how = 'outer')
df = pd.merge(df, df25, how = 'outer')
df = pd.merge(df, df26, how = 'outer')
df = pd.merge(df, df27, how = 'outer')
df = pd.merge(df, df28, how = 'outer')
df = pd.merge(df, df29, how = 'outer')
df = pd.merge(df, df30, how = 'outer')
df

자치구               대여소명     년  월  시간    lambda         mu      p_1+  \
0     송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   0  1.666667   1.533333  0.991495   
1     송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   1  1.033333   1.300000  0.794608   
2     송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   2  0.500000   0.666667  0.749940   
3     송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   3  0.366667   0.333333  0.993272   
4     송파구  롯데월드타워(잠실역2번출구 쪽)  2020  6   4  0.833333   0.466667  1.000000   
...   ...                ...   ... ..  ..       ...        ...       ...   
8635  송파구       천호역 10번 출구 앞  2021  5  19  8.700000  12.966667  0.661627   
8636  송파구       천호역 10번 출구 앞  2021  5  20  6.100000   8.066667  0.734751   
8637  송파구       천호역 10번 출구 앞  2021  5  21  4.066667   5.866667  0.681410   
8638  송파구       천호역 10번 출구 앞  2021  5  22  3.633333   4.366667  0.792371   
8639  송파구       천호역 10번 출구 앞  2021  5  23  3.600000   2.833333  0.964539   

            L_s        W_s        재고량      3사분위수     2사분위수     1사분위수   균형 내 균형  
0     19.336425  12.718886   4.433333  15.941667  5.516667  0.400000   균형   균형  
1      3.837712   3.715147   4.300000  15.941667  5.516667  0.400000   균형   균형  
2      2.993093   5.986662   4.366667  15.941667  5.516667  0.400000   균형   균형  
3     19.951142  60.258851   4.500000  15.941667  5.516667  0.400000   균형   균형  
4     26.727274  57.272733   5.033333  15.941667  5.516667  0.400000   균형  불균형  
...         ...        ...        ...        ...       ...       ...  ...  ...  
8635   1.784030   0.207951  10.566667   9.341667  7.233333  5.016667  불균형   균형  
8636   2.309957   0.389735   8.700000   9.341667  7.233333  5.016667   균형   균형  
8637   1.913942   0.478772   7.033333   9.341667  7.233333  5.016667   균형   균형  
8638   2.827501   0.817192   6.366667   9.341667  7.233333  5.016667   균형   균형  
8639   5.483802   2.006615   7.200000   9.341667  7.233333  5.016667   균형   균형  

[8640 rows x 16 columns]

In [10]:
df_new = df.drop(['자치구', 'lambda', 'mu', '2사분위수'], axis = 1)
df_new = df_new.rename(columns = {'균형' : '기존균형', '내 균형' : 'Queuing균형'})
df_new

대여소명     년  월  시간      p_1+        L_s        W_s  \
0     롯데월드타워(잠실역2번출구 쪽)  2020  6   0  0.991495  19.336425  12.718886   
1     롯데월드타워(잠실역2번출구 쪽)  2020  6   1  0.794608   3.837712   3.715147   
2     롯데월드타워(잠실역2번출구 쪽)  2020  6   2  0.749940   2.993093   5.986662   
3     롯데월드타워(잠실역2번출구 쪽)  2020  6   3  0.993272  19.951142  60.258851   
4     롯데월드타워(잠실역2번출구 쪽)  2020  6   4  1.000000  26.727274  57.272733   
...                 ...   ... ..  ..       ...        ...        ...   
8635       천호역 10번 출구 앞  2021  5  19  0.661627   1.784030   0.207951   
8636       천호역 10번 출구 앞  2021  5  20  0.734751   2.309957   0.389735   
8637       천호역 10번 출구 앞  2021  5  21  0.681410   1.913942   0.478772   
8638       천호역 10번 출구 앞  2021  5  22  0.792371   2.827501   0.817192   
8639       천호역 10번 출구 앞  2021  5  23  0.964539   5.483802   2.006615   

            재고량      3사분위수     1사분위수 기존균형 Queuing균형  
0      4.433333  15.941667  0.400000   균형        균형  
1      4.300000  15.941667  0.400000   균형        균형  
2      4.366667  15.941667  0.400000   균형        균형  
3      4.500000  15.941667  0.400000   균형        균형  
4      5.033333  15.941667  0.400000   균형       불균형  
...         ...        ...       ...  ...       ...  
8635  10.566667   9.341667  5.016667  불균형        균형  
8636   8.700000   9.341667  5.016667   균형        균형  
8637   7.033333   9.341667  5.016667   균형        균형  
8638   6.366667   9.341667  5.016667   균형        균형  
8639   7.200000   9.341667  5.016667   균형        균형  

[8640 rows x 12 columns]

In [11]:
df_new[(df_new['월'] == 5) & (df_new['시간'] == 18)].reset_index(drop = True)

대여소명     년  월  시간      p_1+        L_s       W_s  \
0      롯데월드타워(잠실역2번출구 쪽)  2021  5  18  0.652203   1.875136  0.061346   
1            올림픽공원역 3번출구  2021  5  18  0.973637  11.012676  0.466107   
2               잠실역 8번출구  2021  5  18  0.728743   2.622984  0.172492   
3            한성백제역 2번 출구  2021  5  18  0.874465   5.431544  0.289795   
4           몽촌토성역 1번출구 옆  2021  5  18  0.911078   6.435887  0.337454   
5                  방이삼거리  2021  5  18  0.882030   4.696180  0.292543   
6               방이역 4번출구  2021  5  18  0.927905   4.230818  0.597320   
7            삼전사거리 포스코더샵  2021  5  18  0.970754   5.077110  0.637814   
8            석촌고분역 2번 출구  2021  5  18  0.972180   3.870294  0.667215   
9               석촌동 주민센터  2021  5  18  0.933658   3.857566  0.525212   
10              석촌역 2번출구  2021  5  18  0.905364   3.892766  0.420163   
11              석촌역 8번출구  2021  5  18  0.927512   3.774249  0.500314   
12       석촌호교차로 (스타벅스 앞)  2021  5  18  0.990277   6.418805  0.856629   
13              문정 법조단지8  2021  5  18  0.868272   3.420473  0.444294   
14       문정동 힐스데이트 에코 송파  2021  5  18  0.365942   0.572320  0.113881   
15              문정 법조단지7  2021  5  18  0.495126   0.958200  0.173826   
16            석촌호수 서호사거리  2021  5  18  0.977417   5.878308  0.761282   
17     석촌호수교차로(동호 팔각정 앞)  2021  5  18  0.929004   4.248629  0.566940   
18              송파구청 교차로  2021  5  18  0.639899   1.650751  0.182097   
19          송파나루역 4번 출구옆  2021  5  18  0.990350   9.165266  0.590718   
20              송파역 4번출구  2021  5  18  0.934836   4.792671  0.483656   
21            송파지역자활센터 뒤  2021  5  18  0.986489   6.233224  1.070949   
22  여흥레이크빌 앞 (석촌호수 까페거리)  2021  5  18  0.938490   4.408563  0.563701   
23                 잠실나들목  2021  5  18  0.980576   7.229085  0.597753   
24                잠실나들목4  2021  5  18  0.992010  10.686513  1.009930   
25       잠실나루역 (2번 출구 쪽)  2021  5  18  0.927032   6.446460  0.686237   
26      잠실본동주민센터뒤 잠실근린공원  2021  5  18  0.893780   3.735720  0.524647   
27         잠실새내역 5번 출구 뒤  2021  5  18  0.926086   6.092398  0.435672   
28              잠실역 6번출구  2021  5  18  0.842145   3.372351  0.341291   
29          천호역 10번 출구 앞  2021  5  18  0.909886   4.333928  0.368285   

          재고량      3사분위수     1사분위수 기존균형 Queuing균형  
0    6.333333  15.941667  0.400000   균형        균형  
1   13.600000   6.200000  2.600000  불균형        균형  
2    1.700000   8.566667 -1.675000   균형        균형  
3   12.033333  12.066667  5.566667   균형        균형  
4    8.700000   8.700000  3.733333   균형        균형  
5   16.433333  13.633333  3.925000  불균형        균형  
6    9.166667   7.708333  4.433333  불균형        균형  
7   24.266667  20.400000  8.450000  불균형        균형  
8   23.466667  17.833333  6.100000  불균형        균형  
9   16.333333  13.625000  5.250000  불균형        균형  
10   8.633333   5.650000  3.300000  불균형        균형  
11   5.166667   5.275000  2.166667   균형        균형  
12  17.000000  13.550000  4.708333  불균형        균형  
13   3.800000   3.875000  1.825000   균형        균형  
14  -2.700000   5.700000  0.691667  불균형       불균형  
15  -2.433333   4.175000 -3.841667   균형       불균형  
16  15.100000  15.216667  5.041667   균형        균형  
17   8.666667   7.266667  1.800000  불균형        균형  
18   0.433333   7.175000  0.366667   균형        균형  
19  28.200000  18.083333  6.758333  불균형        균형  
20  10.166667  10.841667  4.658333   균형        균형  
21  30.533333  28.175000  9.833333  불균형        균형  
22  10.100000   8.508333  3.033333  불균형        균형  
23  32.466667  22.025000  9.825000  불균형        균형  
24  19.000000  18.200000  0.000000  불균형        균형  
25   5.133333   7.333333  4.791667   균형        균형  
26  18.166667  22.216667  9.141667   균형        균형  
27  11.400000  11.058333  6.058333  불균형        균형  
28  12.000000   8.875000  4.700000  불균형        균형  
29  14.800000   9.341667  5.016667  불균형        균형

# 기존(월별, 시간별)

# 전체 카이제곱검정

In [110]:
# 분할표

ad_cross = pd.crosstab(df_new['기존균형'], df_new['Queuing균형'])
ad_cross

Queuing균형    균형  불균형
기존균형                
균형         3468  946
불균형        3499  727

In [111]:
chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
chi2, p, dof

(24.726565296461974, 6.606723242509898e-07, 1)

In [112]:
print(f'chi2  = {chi2:.2f}(df={dof:.2f}), p value > {p:.5f}.')

chi2  = 24.73(df=1.00), p value > 0.00000.


# 시간별 카이제곱검정

In [104]:
for i in range(24):
    df_hour = df_new[(df_new['시간'] == i)]
    ad_cross = pd.crosstab(df_hour['기존균형'],  df_hour['Queuing균형'])
    chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
    print(f'chi2  = {chi2:.2f}(df={dof:.5f}), p value > {p:.5f}.')

chi2  = 0.40(df=1.00000), p value > 0.52746.
chi2  = 0.45(df=1.00000), p value > 0.50269.
chi2  = 0.89(df=1.00000), p value > 0.34450.
chi2  = 0.95(df=1.00000), p value > 0.32916.
chi2  = 0.42(df=1.00000), p value > 0.51884.
chi2  = 0.51(df=1.00000), p value > 0.47713.
chi2  = 1.44(df=1.00000), p value > 0.22960.
chi2  = 0.00(df=1.00000), p value > 0.99434.
chi2  = 1.09(df=1.00000), p value > 0.29567.
chi2  = 0.03(df=1.00000), p value > 0.85515.
chi2  = 1.22(df=1.00000), p value > 0.26894.
chi2  = 0.39(df=1.00000), p value > 0.53014.
chi2  = 0.98(df=1.00000), p value > 0.32179.
chi2  = 5.18(df=1.00000), p value > 0.02290.
chi2  = 5.20(df=1.00000), p value > 0.02253.
chi2  = 5.51(df=1.00000), p value > 0.01889.
chi2  = 10.82(df=1.00000), p value > 0.00100.
chi2  = 2.37(df=1.00000), p value > 0.12368.
chi2  = 3.06(df=1.00000), p value > 0.08034.
chi2  = 8.05(df=1.00000), p value > 0.00456.
chi2  = 2.91(df=1.00000), p value > 0.08802.
chi2  = 0.74(df=1.00000), p value > 0.39123.
chi2  = 5

## 18시만

In [105]:
df_hour = df_new[(df_new['시간'] == 18)]
df_hour

대여소명  시간 기존균형 Queuing균형
18       롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
42       롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
66       롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
90       롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
114      롯데월드타워(잠실역2번출구 쪽)  18  불균형        균형
138      롯데월드타워(잠실역2번출구 쪽)  18  불균형        균형
162      롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
186      롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
210      롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
234      롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
258      롯데월드타워(잠실역2번출구 쪽)  18  불균형        균형
282      롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
306            올림픽공원역 3번출구  18  불균형        균형
330            올림픽공원역 3번출구  18  불균형        균형
354            올림픽공원역 3번출구  18   균형        균형
378            올림픽공원역 3번출구  18  불균형        균형
402            올림픽공원역 3번출구  18   균형        균형
426            올림픽공원역 3번출구  18  불균형        균형
450            올림픽공원역 3번출구  18  불균형        균형
474            올림픽공원역 3번출구  18   균형        균형
498            올림픽공원역 3번출구  18  불균형        균형
522            올림픽공원역 3번출구  18  불균형        균형
546            올림픽공원역 3번출구  18  불균형        균형
570            올림픽공원역 3번출구  18  불균형        균형
594               잠실역 8번출구  18  불균형        균형
618               잠실역 8번출구  18  불균형        균형
642               잠실역 8번출구  18   균형        균형
666               잠실역 8번출구  18  불균형        균형
690               잠실역 8번출구  18  불균형        균형
714               잠실역 8번출구  18  불균형        균형
738               잠실역 8번출구  18  불균형       불균형
762               잠실역 8번출구  18   균형        균형
786               잠실역 8번출구  18   균형        균형
810               잠실역 8번출구  18   균형        균형
834               잠실역 8번출구  18   균형        균형
858               잠실역 8번출구  18   균형        균형
882            한성백제역 2번 출구  18  불균형        균형
906            한성백제역 2번 출구  18  불균형        균형
930            한성백제역 2번 출구  18  불균형        균형
954            한성백제역 2번 출구  18   균형        균형
978            한성백제역 2번 출구  18   균형        균형
1002           한성백제역 2번 출구  18   균형        균형
1026           한성백제역 2번 출구  18  불균형        균형
1050           한성백제역 2번 출구  18   균형        균형
1074           한성백제역 2번 출구  18  불균형        균형
1098           한성백제역 2번 출구  18   균형        균형
1122           한성백제역 2번 출구  18  불균형        균형
1146           한성백제역 2번 출구  18   균형        균형
1170          몽촌토성역 1번출구 옆  18  불균형        균형
1194          몽촌토성역 1번출구 옆  18  불균형        균형
1218          몽촌토성역 1번출구 옆  18   균형        균형
1242          몽촌토성역 1번출구 옆  18   균형        균형
1266          몽촌토성역 1번출구 옆  18   균형        균형
1290          몽촌토성역 1번출구 옆  18  불균형        균형
1314          몽촌토성역 1번출구 옆  18  불균형        균형
1338          몽촌토성역 1번출구 옆  18   균형        균형
1362          몽촌토성역 1번출구 옆  18   균형        균형
1386          몽촌토성역 1번출구 옆  18  불균형        균형
1410          몽촌토성역 1번출구 옆  18   균형        균형
1434          몽촌토성역 1번출구 옆  18   균형        균형
1458                 방이삼거리  18   균형        균형
1482                 방이삼거리  18  불균형        균형
1506                 방이삼거리  18  불균형        균형
1530                 방이삼거리  18  불균형        균형
1554                 방이삼거리  18   균형        균형
1578                 방이삼거리  18   균형        균형
1602                 방이삼거리  18   균형        균형
1626                 방이삼거리  18   균형        균형
1650                 방이삼거리  18  불균형        균형
1674                 방이삼거리  18  불균형        균형
1698                 방이삼거리  18  불균형        균형
1722                 방이삼거리  18  불균형        균형
1746              방이역 4번출구  18   균형        균형
1770              방이역 4번출구  18   균형        균형
1794              방이역 4번출구  18   균형        균형
1818              방이역 4번출구  18   균형        균형
1842              방이역 4번출구  18   균형        균형
1866              방이역 4번출구  18   균형        균형
1890              방이역 4번출구  18   균형        균형
1914              방이역 4번출구  18   균형        균형
1938              방이역 4번출구  18   균형        균형
1962              방이역 4번출구  18  불균형        균형
1986              방이역 4번출구  18   균형        균형
2010              방이역 4번출구  18  불균형        균형
2034           삼전사거리 포스코더샵  18   균형        균형
2058           삼전사거리 포스코더샵  18  불균형        균형
2082           삼전사거리

In [106]:
# 분할표

ad_cross = pd.crosstab(df_hour['기존균형'], df_hour['Queuing균형'])
ad_cross

Queuing균형   균형  불균형
기존균형               
균형         165   20
불균형        165   10

In [107]:
chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
chi2, p, dof


(3.05791505791506, 0.08034470154189677, 1)

In [108]:
print(f'chi2  = {chi2:.2f}(df={dof:.5f}), p value > {p:.5f}.')

chi2  = 3.06(df=1.00000), p value > 0.08034.


- 기존 연구 방법과 본 연구 방법을 Two-way 카이제곱검정을 한 결과 p-value가 0.08034로 나와 10% 유의수준하에서 유의한 것으로 나타났다.
- 따라서 기존 연구 방법과 본 연구 방법 간에 연관성이 존재한다.

## 8시, 18시 같이

In [13]:
df_hour = df_new[(df_new['시간'] == 8) | (df_new['시간'] == 18)]
df_hour

대여소명  시간 기존균형 Queuing균형
8     롯데월드타워(잠실역2번출구 쪽)   8   균형        균형
18    롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
32    롯데월드타워(잠실역2번출구 쪽)   8   균형       불균형
42    롯데월드타워(잠실역2번출구 쪽)  18   균형        균형
56    롯데월드타워(잠실역2번출구 쪽)   8   균형        균형
...                 ...  ..  ...       ...
8586       천호역 10번 출구 앞  18  불균형        균형
8600       천호역 10번 출구 앞   8  불균형        균형
8610       천호역 10번 출구 앞  18  불균형        균형
8624       천호역 10번 출구 앞   8   균형        균형
8634       천호역 10번 출구 앞  18  불균형        균형

[720 rows x 4 columns]

In [14]:
# 분할표

ad_cross = pd.crosstab(df_hour['기존균형'], df_hour['Queuing균형'])
ad_cross

Queuing균형   균형  불균형
기존균형               
균형         252   92
불균형        286   90

In [15]:
chi2, p, dof, ef = stats.chi2_contingency(ad_cross, correction = False)
chi2, p, dof


(0.7499360396542218, 0.3864964817496275, 1)

In [16]:
print(f'chi2  = {chi2:.2f}(df={dof:.2f}), p value > {p:.2f}.')

chi2  = 0.75(df=1.00), p value > 0.39.


pvalue가 0.05보다 작아야돼..............